src/code/irqmgr.c

```diff
@@ -1,122 +1,174 @@
+/**
+ * @file irqmgr.c
+ *
+ * This file implements a manager for forwarding three key system interrupt events to
+ * registered clients. These interrupts are:
+ *
+ *  - VI Retrace
+ *      This event is sent to the IRQ manager by the OS VI manager which only supports
+ *      the forwarding of VI events to a single message queue. The IRQ manager will
+ *      forward these events to every registered client. VI retrace events are received
+ *      when the Video Interface has reached the start of the vertical blanking interval,
+ *      happening at approximately 60Hz on NTSC and 50Hz on PAL. Many threads sit idle
+ *      until a VI Retrace event wakes them up, at which point they will perform their
+ *      task and then return to idle to await the next retrace.
+ *
+ *  - Pre-NMI
+ *      This event is sent to the IRQ manager by the OS Interrupt Handler when the reset
+ *      button on the N64 control deck is pressed. This event is forwarded to clients to
+ *      inform them that a reset will occur in 0.5s / 500ms and to begin any shutdown
+ *      procedures.
+ *
+ *  - NMI
+ *      This event is sent at 450ms into the Pre-NMI phase, informing clients that a
```

Comment:
Imo it should be made clearer here that the nmi event does not correspond to the "actual" hardware nmi.
It may also be nice to describe hardware nmi, like the other events have descriptions of retrace and prenmi, even if it doesn't correspond 1:1, to clear up confusion

The event may be more appropriately named as `NMI_SOON`, or something along those lines.
It sounds kind of redundant with PreNMI already being "nmi soon", but it may be a better tradeoff than confusion between NMI and PreNMI+450ms